userspace/libscap/scap.c

```diff
@@ -1169,6 +1169,62 @@ int32_t scap_get_stats(scap_t* handle, OUT scap_stats* stats)
 	return SCAP_SUCCESS;
 }
 
+int scap_get_modifies_state_ppm_sc(uint32_t* ppm_sc_array)
+{
+	if(ppm_sc_array == NULL)
+	{
+		return SCAP_FAILURE;
+	}
+
+#ifdef __linux__
+	// Collect EF_MODIFIES_STATE events
+	for (int i = 0; i < PPM_EVENT_MAX; i++)
+	{
+		if (g_event_info[i].flags & EF_MODIFIES_STATE)
+		{
+			for (int j = 0; j < SYSCALL_TABLE_SIZE; j++)
+			{
+				if (g_syscall_table[j].exit_event_type == i || g_syscall_table[j].enter_event_type == i)
+				{
+					uint32_t ppm_sc_code = g_syscall_code_routing_table[i];
```

Comment:
As per conversations with @FedeDP this might need to be `j` instead of `i` since it iterates on `SYSCALL_TABLE_SIZE`